### ISSUE TITLE
Add rule to detect duplicate (computed) properties

### ISSUE THREAD

**Issue by ajafff** [fimbullinter/wotan#344]
```ts
const A = 'a';
const x = {[A]: 1, [A]: 2}; // duplicate property 'a'
const y = {a: 1, [A]: 2}; // duplicate property 'a'
const z = {[A]: 1, a: 2} // duplicate property 'a'
```

The compiler doesn't emit an error here. See https://github.com/Microsoft/TypeScript/issues/25758

This can probably be merged with `no-duplicate-spread-property` to a more general `no-duplicate-property` rule. Maybe this is already checked, but only on object literals that contain object spread.
